From 233002247cc2675a4ee4beb1cebdafd4b7d7b779 Mon Sep 17 00:00:00 2001
From: "Rywacki, Kamil" <kamil.rywacki@intel.com>
Date: Wed, 8 Jul 2020 10:17:55 +0200
Subject: [PATCH] PONRTSYS-7750: DPM maps ds only gems to omci queue

- Do not set don't care bit (x) on lowest bit of GEM index in lookup table for OMCI queue in GPON mode
- Do not set lookup table for DS only GEM
---
 drivers/net/datapath/dpm/gswip31/datapath_misc.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/net/datapath/dpm/gswip31/datapath_misc.c b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
index 240173498cd2..097a276ca278 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_misc.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_misc.c
@@ -1995,15 +1995,19 @@ static int subif_hw_set(int inst, int portid, int subif_ix,
 		else /*Note:CQM need full subif(15bits)*/
 			lookup.sub_if_id = subif;
 
-		/* For 1st subif and mode 0,
+		/* For mode 0,
 		 * use CBM_QUEUE_MAP_F_SUBIF_DONTCARE,
 		 * otherwise, don't use this flag
 		 */
-		if (!port_info->num_subif &&
-		    port_info->cqe_lu_mode == CQE_LU_MODE0) {
+		if (port_info->cqe_lu_mode == CQE_LU_MODE0) {
 			lookup_f |= CBM_QUEUE_MAP_F_SUBIF_DONTCARE;
 			lookup.sub_if_id_mask_bits = -1;
 		}
+
+		/* Map to drop queue for DS only GEM */
+		if (data->dev->priv_flags & IFF_NO_QUEUE)
+			qid = priv->ppv4_drop_q;
+
 		cbm_queue_map_set(dp_port_prop[inst].cbm_inst, qid,
 				  &lookup, lookup_f);
 		DP_DEBUG(DP_DBG_FLAG_QOS,
